proposals/20220823-huggingface_model_pusher.md

```diff
@@ -44,19 +67,34 @@ HFModelPusher(
-- `repo_url` : repository URL. It is something like f"https://huggingface.co/{repo_id}/{branch}"
+- `commit_id` : the id from the commit history (branch name could be sufficient to retreive a certain version of the model) of the model repository
+- `repo_url` : model repository URL. It is something like f"https://huggingface.co/{repo_id}/{branch}"
+- `space_url` : space repository URL. It is something like f"https://huggingface.co/{repo_id}"
 
-The behaviour of the component:
-1. It pushes the model when the `model` is blessed, or it pushes the `model` when the `model_blessing` parameter is set to `None`. This behaviour inherits from the standard `Pusher` component
-2. Creates HuggingFace Hub Repository object using the `huggingface-hub` package. It will clone one if there is already an existing repository
-3. Checks out a new branch with the name as `pushed_version`. Since the model is pushed for experimental purpose, it would be good to track the versions of the model within separate branches (When the model is ready to be open to public, one can manually merge the right version(branch) into the main branch)
-4. Copy all the model related files into a temporary directory in a local file system. All the model related files produced by the upstream component such as `Trainer`. They could be stored in GCS bucket, so `tf.io.gfile` module is a good choice since it handles files in location agnostic manner (GCS or local)
-5. Add & commit the current status
-6. Pushes the commit to the remote HuggingFace Model Repository
+### HuggingFace Space spacific configurations
+```
+HFSpaceConfig(
+    app_path: str,
+    repo_name: Optional[str],
+    space_sdk: Optional[str] = "gradio",
+    placeholders: Optional[Dict] = None
+)
 
+# default placeholders
+# the keys should be used as is. the values can be 
+# changed as needed. if so, make sure there are the
+# same strings in the files under `app_path`
+{
+    "model_repo": "$HF_MODEL_REPO_NAME",
+    "model_branch": "$HF_MODEL_REPO_BRANCH"
+}
+```
+- `app_path` : path where the application templates are in the container that runs the TFX pipeline. This is expressed either apps.gradio.img_classifier or apps/gradio.img_classifier
+- `repo_name` : the repository name to push the application to. The default value is same as the TFX pipeline name
+- `space_sdk` : either `gradio` or `streamlit`. this will decide which application framework to be used for the Space repository. The default value is `gradio`
+- `placeholders` : dictionary which placeholders to replace with model specific information. The keys represents describtions, and the values represents the actual placeholders to replace in the files under the `app_path`. There are currently two predefined keys, and if `placeholders` is set to `None`, the default values will be used.
 
 ## Project Dependencies
 - [tfx](https://pypi.org/project/tfx/)
```

Comment:
To interact with the model hub it's better to use the [hf_api](https://huggingface.co/docs/huggingface_hub/v0.8.1/en/package_reference).

Reply:
I think hf_api is part of huggingface-hub package?

Reply:
It is. I guess we should list that dependency here?

Reply:
`huggingface-hub` is already listed in the dependency section